### DISCUSSION TITLE
Microsoft SQL datasink

### DISCUSSION THREAD

**Discussion by eckological** [apache/streampipes#1236] [Ideas]
A Microsoft SQL datasink would be very helpful :)

**Comment by tenthe** [comment]
Hey @eckological,
thanks a lot for the suggestion. 
I think that is a great idea and my suggestion would be to direclty transfer this discussion to an issue and mark it as a `good first issue`

**Comment by bossenti** [comment]
@eckological have you any features in mind that such a sink should definitely provide?

**Reply by eckological** [reply]
Sorry, nothing except the obvious, Hostname, port, database name, table name, credentials, SSL

**Reply by eckological** [reply]
Maybe some way to use stored procedures in case you want to do something more fancy than "just insert into a table"?

**Reply by bossenti** [reply]
For an initial version the first version sounds realistic
For the second one I cannot really access how much effort this would mean.
So I would suggest to create an issue for the sink with the feature set described in the first comment.

Would you be interested in contribution here @eckological? 🙂 
Please don't feel obliged, just curious if you are interested in
Of course we would provide some mentoring in case

**Reply by eckological** [reply]
@bossenti I have played around with the idea for a couple of weeks and i am intrigued. However, i'm not proficient in Java. I have some experience with Python and C#.

**Reply by bossenti** [reply]
@eckological 
Ok, then we will create an issue for this sink. Hopefully, we can work on it soon 🙂 

Interesting, then our brand-new python library might be interesting for you which we will release the first time with our upcoming version `0.91.0` 💫 
Stay tuned!

**Reply by bossenti** [reply]
issue is created https://github.com/apache/streampipes/issues/1348

**Comment by flomickl** [comment]
Can Microsoft SQL connect via jdbc driver?

Does Micrsofot SQL have different data types than already implemented here 
https://github.com/apache/streampipes/blob/dev/streampipes-extensions/streampipes-sinks-databases-jvm/src/main/java/org/apache/streampipes/sinks/databases/jvm/jdbcclient/model/DbDataTypes.java


and handled here https://github.com/apache/streampipes/blob/dev/streampipes-extensions/streampipes-sinks-databases-jvm/src/main/java/org/apache/streampipes/sinks/databases/jvm/jdbcclient/model/DbDataTypeFactory.java

?

**Reply by bossenti** [reply]
These are indeed important aspects @flomickl ❗ 
I would propose to sort them out during working on the implementation of the issue or in a dedicated dev-discussion beforehand.
Would you agree? 🙂

**Comment by flomickl** [comment]
@bossenti 

I think we should do both ^^
The summarized conclusion for the Microsoft SQL here.

In general, we should discuss and document the JDBC database drivers and types.
My plan is to create a postgis sink soon anyway and maybe other db sinks. So, a good moment I think

Ages Ago I started this section https://cwiki.apache.org/confluence/x/Qg54CQ
before you have rewritten the JDBC stuff :)

We should update and fill the blank spots.
What do you think?

**Reply by bossenti** [reply]
That's a valid point.
My intention was just to carve out all the technical stuff from the feature request discussion to a dedicated one.

Looks like you already spent a lot of conceptual work on our JDBC setup 👍🏼 
So we are probably talking about harmonizing and refactoring our JDBC sinks or are you more about documenting the status quo?

**Reply by flomickl** [reply]
I am uncertain if we are talking about harmonizing and refactoring our JDBC
This was way before you did https://github.com/apache/streampipes/issues/512

So let's have a look. But we should discuss this in a separate discussion here  #1313

**Reply by bossenti** [reply]
Thanks for creating the discussion 🙏🏼

**Reply by flomickl** [reply]
I also added a general discussion about sinks structure  #1314 
I think for the Microsoft SQL sink implementation, the outcome of the 1 class method and type is relevant :)
